react-color/index.d.ts

```diff
@@ -1,249 +1,58 @@
-declare module "react-color/lib/components/circle/Circle" { export default ReactColor.CirclePicker }
-declare module "react-color/lib/components/compact/Compact" { export default ReactColor.CompactPicker }
-declare module "react-color/lib/components/github/Github" { export default ReactColor.GithubPicker }
-declare module "react-color/lib/components/hue/Hue" { export default ReactColor.HuePicker }
-declare module "react-color/lib/components/meterial/Material" { export default ReactColor.MaterialPicker }
-declare module "react-color/lib/components/photoshop/Photoshop" { export default ReactColor.PhotoshopPicker }
-declare module "react-color/lib/components/sketch/Sketch" { export default ReactColor.SketchPicker }
-declare module "react-color/lib/components/slider/Slider" { export default ReactColor.SliderPicker }
-declare module "react-color/lib/components/swatches/Swatches" { export default ReactColor.SwatchesPicker }
-declare module "react-color/lib/components/twitter/Twitter" { export default ReactColor.TwitterPicker }
+export type ColorChangeHandler = (color: ColorResult) => void;
 
-declare module "react-color" {
-    import AlphaPicker from "react-color/lib/components/alpha/Alpha"
-    import BlockPicker from "react-color/lib/components/block/Block"
-    import ChromePicker from "react-color/lib/components/chrome/Chrome"
-    import CirclePicker from "react-color/lib/components/circle/Circle"
-    import CompactPicker from "react-color/lib/components/compact/Compact"
-    import GithubPicker from "react-color/lib/components/github/Github"
-    import HuePicker from "react-color/lib/components/hue/Hue"
-    import MaterialPicker from "react-color/lib/components/meterial/Material"
-    import PhotoshopPicker from "react-color/lib/components/photoshop/Photoshop"
-    import SketchPicker from "react-color/lib/components/sketch/Sketch"
-    import SliderPicker from "react-color/lib/components/slider/Slider"
-    import SwatchesPicker from "react-color/lib/components/swatches/Swatches"
-    import TwitterPicker from "react-color/lib/components/twitter/Twitter"
-    import CustomPicker from "react-color/lib/components/common/ColorWrap"
-
-    export type CustomPickerProps<A> = ReactColor.CustomPickerProps<A>
+export interface ColorPickerProps<A> extends ClassAttributes<A> {
+    color?: Color;
+    onChange?: ColorChangeHandler;
+    onChangeComplete?: ColorChangeHandler;
+}
 
-    export {
-        AlphaPicker,
-        BlockPicker,
-        ChromePicker,
-        CirclePicker,
-        CompactPicker,
-        GithubPicker,
-        HuePicker,
-        MaterialPicker,
-        PhotoshopPicker,
-        SketchPicker,
-        SliderPicker,
-        SwatchesPicker,
-        TwitterPicker,
-        CustomPicker
-    }
+export interface CustomPickerProps<A> extends ClassAttributes<A> {
+    color?: Color;
+    pointer?: ReactNode;
+    onChange: ColorChangeHandler;
```

Comment:
This interface is used in `Saturation`, `Alpha` and `Hue` common components and after checking the source code it seems they all require `onClick` handler specified. I created and [issue in their repo](https://github.com/casesandberg/react-color/issues/327) to make the handler optional. Right now the check if property has been provided as a function is not present. Not specifying `onClick` will cause errors in all of the components.

Here is source code reference:

- `Saturation` [source code](https://github.com/casesandberg/react-color/blob/master/src/components/common/Saturation.js#L10-L25)
- `Alpha` [source code](https://github.com/casesandberg/react-color/blob/master/src/components/common/Alpha.js#L12-L15)
- `Hue` [source code](https://github.com/casesandberg/react-color/blob/master/src/components/common/Hue.js#L10-L13)